balancer-labs/frontend-v2
# Tokenomics - Staking Contract Implementation

# Description

This pull request integrates the staking contracts and gauge subgraph into the home page.
Very briefly: 
1. There is a new pool on the home page which shows the pools the user is staked in, or can stake in
2. Adds the ability to approve and stake your BPT right from the home page
3. Separates migrateable pools into their own table

## Type of change


## How should this be tested?

1. Load up the home page
2. Make sure all tables are loading appropriately
3. Ensure you have no staked pools, invest into one that has a gauge. (Here is one `0x3a19030ed746bd1c3f2b0f996ff9479af04c5f0a000200000000000000000004`)
4. Once you have invested, the pool should show up under 'My Staked Pools'
5. Given the fact that you have not staked anything, the 'Stake' button should be visible
6. Go through the staking process by clicking the 'stake' button. It should allow you to approve and stake appropriately. 
7. After staking, make sure that the table updates to show that you have 100% staked, no button should be visible.
8. Now invest some more funds into the same pool, the table should update to show the 'stake' button again

## Visual context

https://www.loom.com/share/1532079e0ed249c6a1ac6dc131fa748a

## Checklist:
